lib/incito-browser/incito.js

```diff
@@ -1,308 +1,696 @@
         this.incito = incito;
-        this.renderLaziness = renderLaziness;
         this.el = document.createElement('div');
+        this.imageCount = 0;
+        this.preloadImageCount = 20;
         this.ids = {};
-        this.views = flattenViews([], this.incito.root_view);
-        this.viewsLength = this.views.length;
-        this.viewIndex = 0;
-        this.lazyloadables = [];
-        this.lazyloader = throttle(this.lazyload.bind(this), 150);
-        this.renderedOutsideOfViewport = false;
-        this._events = {};
+        this.sections = [];
+        this.shouldLazyload = 'IntersectionObserver' in window;
     }
 
-    bind(event, fn) {
-        this._events[event] = this._events[event] || [];
-        return this._events[event].push(fn);
-    }
+    start() {
+        loadFonts(this.incito.font_assets);
+
+        const html = this.renderHtml();
+        const theme = this.incito.theme || {};
+
+        this.el.dataset.readme = 'Incito by Tjek (https://incito.io)';
+        this.el.className = 'incito';
+
+        if (Array.isArray(theme.font_family)) {
+            this.el.style.fontFamily = theme.font_family.join(', ');
+        }
+
+        if (isDefinedStr(theme.background_color)) {
+            this.el.style.backgroundColor = theme.background_color;
+        }
+
+        if (isDefinedStr(theme.text_color)) {
+            this.el.style.color = theme.text_color;
+        }
+
+        if (typeof theme.line_spacing_multiplier === 'number') {
+            this.el.style.lineHeight = theme.line_spacing_multiplier;
+        }
+
+        // By setting the language we help the browser with stuff like hyphenation.
+        if (this.incito.locale != null) {
+            this.el.setAttribute('lang', this.incito.locale);
+        }
 
-    unbind(event, fn) {
-        if (this._events[event]) {
-            return this._events[event].splice(
-                this._events[event].indexOf(fn),
-                1
+        this.el.innerHTML = html;
+
+        this.el.addEventListener('click', (e) => {
+            const link = e.target.getAttribute('data-link');
+
+            if (isDefinedStr(link)) {
+                window.open(link, '_blank');
+            }
+        });
+
+        this.containerEl.appendChild(this.el);
+
+        if (this.shouldLazyload) {
+            this.lazyloader = new IntersectionObserver(
+                (entries) => {
+                    entries.forEach((entry) => {
+                        if (entry.isIntersecting) {
+                            this.loadEl(entry.target);
+                            this.lazyloader.unobserve(entry.target);
+                        }
+                    });
+                },
+                {
+                    rootMargin: '500px'
+                }
             );
+
+            this.el.querySelectorAll('.incito--lazy').forEach((lazyEl) => {
+                this.lazyloader.observe(lazyEl);
+            });
         }
+
+        this.trigger('allRendered');
```

Comment:
Since this runs totally synchronously now we can ditch this event, knowing that after `Incito.start()` everything will have been rendered.

Reply:
I just wasn't sure if allRendered was needed by someone?

Reply:
I'll remove it.

Reply:
Not to my knowledge, but it's very hard for us to know. Easy fix if anyone was though, just move `allRendered` event handler code after `viewer.start()`

Introduced here btw https://github.com/shopgun/incito-browser/commit/ff28c61383b829f6a8affc5befbd7d19e984f21a